content/source/docs/extend/best-practices/testing.html.md

```diff
@@ -159,8 +159,11 @@ func testAccCheckExampleResourceExists(n string, widget *example.Widget) resourc
 
 		// If no error, assign the response Widget attribute to the widget pointer
 		*widget = *resp.Widget
+		if *widget == nil {
```

Comment:
`*widget` is dereferencing the pointer, so it can never be `nil`. This would actually panic on line 161 if `resp.Widget` were `nil`, as `*resp.Widget` is trying to dereference it. The `nil` check and error is probably better located before that, on line 159.

Reply:
@paddycarver Thanks for the helpful comments. I have changed the PR, please check if the changes reflect your comments correctly.